Mathlib/Data/Finset/Sups.lean

```diff
@@ -178,6 +176,21 @@ theorem subset_sups {s t : Set α} :
   subset_image₂
 #align finset.subset_sups Finset.subset_sups
 
+lemma image_sups (f : F) (s t : Finset α) : image f (s ⊻ t) = image f s ⊻ image f t :=
+  image_image₂_distrib $ map_sup f
+
+lemma map_sups (f : F) (hf) (s t : Finset α) :
+    map ⟨f, hf⟩ (s ⊻ t) = map ⟨f, hf⟩ s ⊻ map ⟨f, hf⟩ t := by
+  simpa [map_eq_image] using image_sups f s t
+
+lemma subset_sups_self : s ⊆ s ⊻ s := fun _a ha ↦ mem_sups.2 ⟨_, ha, _, ha, sup_idem⟩
+lemma sups_self_subset : s ⊻ s ⊆ s ↔ SupClosed (s : Set α) := sups_subset_iff
+@[simp] lemma sups_self : s ⊻ s = s ↔ SupClosed (s : Set α) := by simp [←coe_inj]
```

Comment:
To distinguish these from the lemma above, I think they should end in `_iff`

Reply:
(and for the `Set` and `infs` versions)

Reply:
I'll go with `sups_self_eq`